python/cudf/cudf/core/indexed_frame.py

```diff
@@ -867,3 +867,66 @@ def resample(
             if isinstance(self, cudf.Series)
             else cudf.core.resample.DataFrameResampler(self, by=by)
         )
+
+    def _reset_index(self, level, drop, col_level=0, col_fill=""):
+        """Shared path for DataFrame.reset_index and Series.reset_index."""
+        if col_level != 0:
+            raise NotImplementedError(
+                "col_level parameter is not supported yet."
+            )
+
+        if col_fill != "":
+            raise NotImplementedError(
+                "col_fill parameter is not supported yet."
+            )
+
+        if level is not None and not isinstance(level, (tuple, list)):
+            level = (level,)
+        _check_duplicate_level_names(level, self._index.names)
+
+        # Split the columns in the index into data and index columns
+        (
+            data_columns,
+            index_columns,
+            column_names,
+            index_names,
+        ) = self._index._split_columns_by_levels(level)
+        if index_columns:
+            index = _index_from_data(
+                dict(zip(range(len(index_columns)), index_columns)),
```

Comment:
I've seen this pattern crop up enough times that I think it's worth defining an additional helper. Something like:
```
def _index_from_column_list(columns):
    return _index_from_data(dict(zip(range(len(index_columns)), index_columns))))
```

Reply:
I'm not sure about the "enough times" part, the other place I see it is in `_from_columns`. But I think giving it a name is more readable.

Reply:
There's a question about single level index though. We name the keys in dict with number sequence to support duplicate level names for multiindex. This isn't really necessary for single level index. So using this function with `len(columns) == 1` is probably an anti-pattern. Should we raise an error/warning here.?

Reply:
I see your point, but I'm not really sure what the right solution is here. The point of code paths like this would be for them to work whether you get a MultiIndex or a single columned index. Having to check that beforehand seems like it defeats the purpose of `_index_from_data` support building a MultiIndex, for instance.

Reply:
I think we can introduce the function, document it well, and revisit this if anti pattern do show up.